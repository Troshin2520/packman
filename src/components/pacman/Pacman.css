.pacman{
    width: 1.5vw;
    height: 1.5vw;
    z-index: 2;
    font-size: 1vw;
}

.pacman-top{
    background-color: yellow;
    height: 0.75vw;
    width: 1.5vw;
    border-radius: 0.75vw 0.75vw 0 0;
    animation: spin1 0.5s infinite linear;
    z-index: 3;
    text-align: center;
}

.pacman-bottom{
    background-color: yellow;
    height: 0.75vw;
    width: 1.5vw;
    border-radius: 0 0 0.75vw 0.75vw;
    animation: spin2 0.5s infinite linear;
    z-index: 3;
}

.pacman-turn-top {
    transform: rotate(270deg);
}

@keyframes spin2 {
    0%  {transform: rotate(0deg);}
    50%{transform: rotate(35deg);}
}


@keyframes spin1 {
    0%  {transform: rotate(0deg);}
    50%{transform: rotate(-35deg);}
}
