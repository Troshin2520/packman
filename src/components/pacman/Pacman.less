.pacman {
  width: 3rem;
  height: 3rem;
  z-index: 2;
  font-size: 2rem;

  &-top {
    background-color: gold;
    height: 1.5rem;
    width: 3rem;
    border-radius: 1.5rem 1.5rem 0 0;
    animation: spin-top 0.5s infinite linear;
    z-index: 3;
    text-align: center;
  }

  &-bottom {
    background-color: gold;
    height: 1.5rem;
    width: 3rem;
    border-radius: 0 0 1.5rem 1.5rem;
    animation: spin-bottom 0.5s infinite linear;
    z-index: 3;
  }

  &-turn-top {
    transform: rotate(270deg);
  }
}

@keyframes spin-bottom {
  0% {
    transform: rotate(0deg);
  }
  50% {
    transform: rotate(35deg);
  }
}

@keyframes spin-top {
  0% {
    transform: rotate(0deg);
  }
  50% {
    transform: rotate(-35deg);
  }
}
