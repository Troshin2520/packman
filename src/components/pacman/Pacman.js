import React, {Component} from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import './Pacman.less';
import {ACTION_MOVE_PACMAN, BLOCK_SIZE} from '../../constants';

class Pacman extends Component {

  constructor(props) {
    super(props);
  }

  shouldComponentUpdate() {
    return true;
  }

  onAnimationEnd() {

  }

  render() {
    return (<div className="pacman"
                 style={{top:`${this.props.y * BLOCK_SIZE}rem`, left:`${this.props.x * BLOCK_SIZE}rem`}}
                 onAnimationEnd={this.onAnimationEnd}>
              <div className="top">&bull;</div>
              <div className="bottom"></div>
            </div>);
  }
}

Pacman.propTypes = {
  x: PropTypes.number,
  y: PropTypes.number
};

Pacman.defaultProps = {
  x: 0,
  y: 0
};

export default connect(
  state => state.pacman,
  dispatch => ({
    onChangeState(type, item) {
      dispatch({type: type, payload: item});
    }
  })
)(Pacman);