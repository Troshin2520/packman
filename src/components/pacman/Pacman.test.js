import React from 'react';
import Provider from 'redux';
import Renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Pacman from './Pacman';

describe('Pacman tests', () => {

  it('Pacman rendered without crushing', () => {

    const initialState = {pacman:{x: 6, y: 8, dir: 'left'}};
    const mockStore = configureStore();
    const store = mockStore(initialState);

    const component = Renderer.create(
        <Pacman store={store} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
