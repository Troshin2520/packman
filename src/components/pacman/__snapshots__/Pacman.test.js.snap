// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pacman tests Pacman rendered without crushing 1`] = `
<div
  className="pacman"
  onAnimationEnd={[Function]}
  style={
    Object {
      "left": "24rem",
      "top": "32rem",
    }
  }
>
  <div
    className="top"
  >
    â€¢
  </div>
  <div
    className="bottom"
  />
</div>
`;
